apps/src/aichat/views/tabs/Tabs.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {useState} from 'react';
+import classNames from 'classnames';
+
+import styles from './tabs.module.scss';
+import TabPanel from './TabPanel';
+
+type Tab = {
+  title: string;
+  content: React.ReactNode;
+};
+
+interface TabsProps {
+  tabs: Tab[];
+  // Identifier used for ARIA properties
+  // to differentiate between uses of this component on the page.
+  name: string;
+  initialActiveIndex?: number;
+}
+
+const Tabs: React.FunctionComponent<TabsProps> = ({
+  tabs,
+  name,
+  initialActiveIndex = 0,
+}) => {
+  const [activeIndex, setActiveIndex] = useState<number>(
+    initialActiveIndex >= 0 && initialActiveIndex < tabs.length
+      ? initialActiveIndex
+      : 0
+  );
+
+  const handleTabClick = (index: number) => {
+    setActiveIndex(index);
+  };
+
+  const nameStripped = name.replace(' ', '-');
+  const getTabButtonId = (index: number) => `${nameStripped}-tab-${index}`;
```

Comment:
nit: use `useMemo` for `getTabPanelId` and `getTabButtonId` here?

Reply:
I don't think this is an expensive enough calculation to make `useMemo` worthwhile?

Reply:
+1, fair enough